src/operations/blas3/gemm_no_local_full_vec.hpp

```diff
@@ -0,0 +1,911 @@
+    const index_t k = a_.get_size_col();
+    const index_t lda = a_.getSizeL();
+    const index_t ldb = b_.getSizeL();
+    const index_t ldc = c_.getSizeL();
+
+    // The batch index that each workgroup should start working with
+    const index_t wg_batch_id = id.get_group(0) / get_workgroup_cluster();
+    // This will disable all workgroups that dont have any batch to work on
+    if (wg_batch_id >= batch_size_) {
+      return;
+    }
+
+    const index_t batch_stride =
+        id.get_group_range(0) / get_workgroup_cluster();
+
+    const index_t a_size = trans_a ? m * lda : k * lda;
+    const index_t b_size = trans_b ? ldb * k : n * ldb;
+    const index_t c_size = ldc * n;
+
+    auto orig_A = a_.get_pointer() + (wg_batch_id * a_size);
+    auto orig_B = b_.get_pointer() + (wg_batch_id * b_size);
+    auto orig_C = c_.get_pointer() + (wg_batch_id * c_size);
+
+    const index_t number_of_block_per_row = ((m - 1) / block_rows) + 1;
+    /* linear work group id The number of work-group required to executed each
+     * batch efficiently*/
+    const index_t wg_id = id.get_group(0) % get_workgroup_cluster();
+    /* linear work item id */
+    const index_t item_id = id.get_local_id(0);
+    /* row tile id  per work group */
+    const index_t tile_id_row = wg_id % number_of_block_per_row;
+    /* column tile id per work group */
+    const index_t tile_id_col = wg_id / number_of_block_per_row;
+    /* the start position of the tile-row per work group */
+    const index_t wg_row = tile_id_row * block_rows;
+    /* the start position of the tile-column per work group */
+    const index_t wg_col = tile_id_col * block_cols;
+    /*!
+     * @brief is_internal_block is used to distinguish
+     * the internal block. Therefore, work items using these blocks don't need
+     * to check for boundaries. Checking the packet size is a workaround because
+     * normally the vector size and item rows/cols must all be equal, but when
+     * vectorization is disabled the vector size is always 1 and the algorithm
+     * breaks.
+     */
+    const bool is_internal_block =
+        (packetize_t::packet_size !=
+         1) &&  // This is a workaround when vectorization is disabled.
+        (m - wg_row >= block_rows) &&
+        (n - wg_col >= block_cols);
+
+    const index_t vector_ofs = is_internal_block ? packetize_t::packet_size : 1;
+    /* work item id per row */
+    const index_t local_item_id_row = item_id % wg_rows * vector_ofs;
+    /* work item id per column */
+    const index_t local_item_id_col =
+        (item_id / wg_rows) * (trans_b ? vector_ofs : 1);
+
+    /* Exiting from any threads outside of the m and n boundary */
+    const bool out_of_range = ((local_item_id_row + wg_row >= m) ||
+                               (local_item_id_col + wg_col >= n));
+
+    m -= local_item_id_row + wg_row;
+    n -= local_item_id_col + wg_col;
+    /*
+     * The ma and nb are used to adjust the start position of each work-item for
```

Comment:
This comment doesn't look to correspond to the code. Should `ma` actually by `dim_m_a_start`?

Reply:
Fixed!